src/engine.rs

```diff
@@ -441,7 +456,16 @@ impl Reedline {
         loop {
             let mut paste_enter_state = false;
 
-            if event::poll(Duration::from_millis(1000))? {
+            #[cfg(feature = "external_printer")]
+            if let Some(ref external_printer) = self.external_printer {
+                if let Ok(line) = external_printer.receiver().try_recv() {
+                    self.painter.print_crlf()?;
+                    self.print_line(&line)?;
+                    self.repaint(prompt)?;
```

Comment:
`repaint()` it self is now probalby pretty robust to what state the editor is in. But if the prompt origin position is fixed and the printed output scrolls the screen, stuff might shift around and things get overprinted.

Reply:
This is what I was afraid of... I have to test it more by hand, but until now, all looks good, on my machine ;)